From 0e1416e65c99aca947e589bfa56d5bc832c023d6 Mon Sep 17 00:00:00 2001
From: Lukas Slebodnik <lslebodn@redhat.com>
Date: Sat, 27 May 2017 14:39:45 +0200
Subject: [PATCH 165/166] SECRETS: Fix warning Wpointer-bool-conversion

Debug messages would always say that verify_peer and verify_host
are enabled. Even though they would be explicitly disabled.

src/responder/secrets/proxy.c:143:18: error:
    address of 'cfg->verify_peer' will always evaluate to
      'true' [-Werror,-Wpointer-bool-conversion]
          (&cfg->verify_peer ? "true" : "false"));
            ~~~~~^~~~~~~~~~~ ~
src/util/debug.h:108:32: note: expanded from macro 'DEBUG'
                     format, ##__VA_ARGS__); \
                               ^~~~~~~~~~~
src/responder/secrets/proxy.c:149:18: error:
    address of 'cfg->verify_host' will always evaluate to
      'true' [-Werror,-Wpointer-bool-conversion]
          (&cfg->verify_host ? "true" : "false"));
            ~~~~~^~~~~~~~~~~ ~
src/util/debug.h:108:32: note: expanded from macro 'DEBUG'
                     format, ##__VA_ARGS__); \
                               ^~~~~~~~~~~

Reviewed-by: Jakub Hrozek <jhrozek@redhat.com>
---
 src/responder/secrets/proxy.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/responder/secrets/proxy.c b/src/responder/secrets/proxy.c
index 9c2aa425d414728d10aa830f640632e98def3c1c..a4e97f83ef406e71a1e6509a6b719c47afdfd5b8 100644
--- a/src/responder/secrets/proxy.c
+++ b/src/responder/secrets/proxy.c
@@ -140,13 +140,13 @@ static int proxy_sec_get_cfg(struct proxy_context *pctx,
                           true, &cfg->verify_peer);
     if (ret) goto done;
     DEBUG(SSSDBG_CONF_SETTINGS, "verify_peer: %s\n",
-          (&cfg->verify_peer ? "true" : "false"));
+          cfg->verify_peer ? "true" : "false");
 
     ret = confdb_get_bool(pctx->cdb, secreq->cfg_section, "verify_host",
                           true, &cfg->verify_host);
     if (ret) goto done;
     DEBUG(SSSDBG_CONF_SETTINGS, "verify_host: %s\n",
-          (&cfg->verify_host ? "true" : "false"));
+          cfg->verify_host ? "true" : "false");
 
     ret = proxy_get_config_string(pctx, cfg, false, secreq,
                                   "capath", &cfg->capath);
-- 
2.9.4

